bindings/c/foundationdb/fdb_c.h

```diff
@@ -225,6 +228,9 @@ extern "C" {
     fdb_transaction_get_committed_version( FDBTransaction* tr,
                                            int64_t* out_version );
 
+    DLLEXPORT WARN_UNUSED_RESULT FDBFuture*
+    fdb_transaction_get_approximate_size(FDBTransaction* tr);
```

Comment:
We'd probably have to be careful about doing something like that. One option would be to block on the result and return it when done, but we'd have to make sure the implementation works when called from the main thread (or is implemented such that a call from the main thread is impossible).

Another alternative is to bypass the main thread, perhaps by using an atomic type. One risk here, though, is that it may be possible to call this on a transaction that hasn't been constructed yet (i.e. that work is pending on the main thread). It also would potentially behave unexpectedly by excluding recent operations.

Reply:
How does this work with our other operations that don't return a future (like getting the commit version)?

Reply:
For that function in particular, it is required that you call commit and wait for the result before reading the committed version.

Reply:
What about setting a key or applying a mutation? It seems like those would be susceptible to more or less the same problems if we would need to worry about things like excluding recent operations in this case.

Reply:
Setting a key is done on the main thread, but we don't wait on the completion before returning. That is in part why bypassing the main thread for approximate size could be weird -- it's possible that none of your mutations will have been counted yet.

Reply:
Ah, I see.

Well, it still seems somewhat of a shame given that I think (with the kind of weird exception of `getVersionstamp`) that all of our operations that do I/O return a future, and all of the ones that don't don't. But I can see why you'd want this number to include all previous updates. It seems like we should document why it's returning a future, though.

Reply:
I've added a comment for documenting why `fdb_transaction_get_approximate_size` returns a future.